server/schedule/opt/healthy.go

```diff
@@ -54,7 +54,13 @@ func HealthAllowPending(cluster Cluster) func(*core.RegionInfo) bool {
 
 // AllowBalanceEmptyRegion returns a function that checks if a region is an empty region and can be balanced.
 func AllowBalanceEmptyRegion(cluster Cluster) func(*core.RegionInfo) bool {
-	return func(region *core.RegionInfo) bool { return IsEmptyRegionAllowBalance(cluster, region) }
+	switch cluster.GetClusterType() {
```

Comment:
Can we use type assertion?

Reply:
it will introduce cycle import between `schedule`  and `schedule/opt`